nilearn/masking.py

```diff
@@ -513,6 +513,137 @@ def compute_multi_background_mask(data_imgs, border_size=2, upper_cutoff=0.85,
     return mask
 
 
+def compute_gray_matter_mask(target_img, threshold=.5,
+                             connected=True, opening=2, memory=None,
```

Comment:
May be you need to think of adding an argument called ```ensure_finite``` ? This deals with NaNs in the images.

Reply:
I don't use the data values of `target_img`, but only its shape and affine, would it be relevant to check the values anyhow ?